src/DotVVM.Framework/ViewModel/Serialization/DotvvmDictionaryConverter.cs

```diff
@@ -0,0 +1,80 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using DotVVM.Framework.Utils;
+using Newtonsoft.Json;
+
+namespace DotVVM.Framework.ViewModel.Serialization
+{
+    public class DotvvmDictionaryConverter : JsonConverter
+    {
+        public override void WriteJson(JsonWriter writer, object value, JsonSerializer serializer)
+        {
+            if (value == null)
+            {
+                writer.WriteNull();
+            }
+            else
+            {
+                var dict = value as IDictionary;
+
+                var attrs = value.GetType().GetGenericArguments();
+                var keyValuePair = typeof(KeyValuePair<,>).MakeGenericType(attrs);
+                var listType = typeof(List<>).MakeGenericType(keyValuePair);
+
+
+                var keyEnumerator = dict.Keys.GetEnumerator();
+                var valuesEnumerator = dict.Values.GetEnumerator();
```

Comment:
I don't think there is a guarantee that the order of keys and values must be the same. Why not dict.GetEnumerator?

Reply:
I just didn't want to use another reflection, but it makes sense to use just dict.GetEnumerator